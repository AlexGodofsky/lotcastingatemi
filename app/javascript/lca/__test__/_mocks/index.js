export const mockCharacter = {
  type: 'Character',
  attr_dexterity: 3, attr_wits: 3, attr_manipulation: 3, attr_stamina: 3, attr_strength: 2,
  abil_dodge: 4,  abil_integrity: 4, abil_socialize: 4,
  abil_melee: 2,
  abil_craft: [], abil_martial_arts: [], specialties: [],
  armor_weight: 'unarmored', armor_is_artifact: false,
  health_level_0s: 1, health_level_1s: 2, health_level_2s: 2, health_level_4s: 1, health_level_incap: 1,
  damage_bashing: 0, damage_lethal: 0, damage_aggravated: 0
}
