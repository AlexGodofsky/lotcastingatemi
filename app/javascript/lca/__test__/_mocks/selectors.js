// @flow
import { getPoolsAndRatings, getPenalties } from '../../selectors'
import type { fullChar } from '../../utils/propTypes/flow.js'

export const mockStateForCharacter = (character: fullChar) => ({ entities: {
  current: {
    characters: {
      [0]: {
        id: 0,
        ...character,
        merits: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
        weapons: [1, 2, 3],
        spells: [1, 2, 3],
      }
    },
    merits: {
      [1]:  { merit_name: 'ambidextrous' },
      [2]:  { merit_name: 'danger sense' },
      [3]:  { merit_name: 'fast reflexes' },
      [4]:  { merit_name: 'fleet of foot' },
      [5]:  { merit_name: 'mighty thew', rating: 2 },
      [6]:  { merit_name: 'pain tolerance' },
      [7]:  { merit_name: 'toxin resistance' },
      [8]:  { merit_name: 'chameleon' },
      [9]:  { merit_name: 'enhanced sense' },
      [10]: { merit_name: 'unusual hide', rating: 2 },
      [11]: { merit_name: 'living spirit cultivation' },
      [12]: { merit_name: 'vital focus cultivation' },
      [13]: { merit_name: 'hideous' },
    },
    spells: {
      [1]:  { name: 'invulnerable skin of bronze', control: true },
      [2]:  { name: 'impenetrable veil of night', control: true },
      [3]:  { name: 'rain of doom', control: true },
    },
    weapons: {
      [1]:  { name: 'light weapon',  weight: 'light',  attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
      [2]:  { name: 'medium weapon', weight: 'medium', attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
      [3]:  { name: 'heavy weapon',  weight: 'heavy',  attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
    },
    charms: {
      [1]:  { ability: 'archery',       keywords: ['excellency'] },
      [2]:  { ability: 'athletics',     keywords: ['excellency'] },
      [3]:  { ability: 'awareness',     keywords: ['excellency'] },
      [4]:  { ability: 'brawl',         keywords: ['excellency'] },
      [5]:  { ability: 'bureaucracy',   keywords: ['excellency'] },
      [6]:  { ability: 'craft',         keywords: ['excellency'] },
      [7]:  { ability: 'dodge',         keywords: ['excellency'] },
      [8]:  { ability: 'integrity',     keywords: ['excellency'] },
      [9]:  { ability: 'investigation', keywords: ['excellency'] },
      [10]: { ability: 'larceny',       keywords: ['excellency'] },
      [11]: { ability: 'linguistics',   keywords: ['excellency'] },
      [12]: { ability: 'lore',          keywords: ['excellency'] },
      [13]: { ability: 'medicine',      keywords: ['excellency'] },
      [14]: { ability: 'melee',         keywords: ['excellency'] },
      [15]: { ability: 'occult',        keywords: ['excellency'] },
      [16]: { ability: 'performance',   keywords: ['excellency'] },
      [17]: { ability: 'presence',      keywords: ['excellency'] },
      [18]: { ability: 'resistance',    keywords: ['excellency'] },
      [19]: { ability: 'ride',          keywords: ['excellency'] },
      [20]: { ability: 'sail',          keywords: ['excellency'] },
      [21]: { ability: 'socialize',     keywords: ['excellency'] },
      [22]: { ability: 'stealth',       keywords: ['excellency'] },
      [23]: { ability: 'survival',      keywords: ['excellency'] },
      [24]: { ability: 'thrown',        keywords: ['excellency'] },
      [25]: { ability: 'war',           keywords: ['excellency'] },
      [26]: { charm_type: 'MartialArts' },
    }
  }
}})

export const mockGetPoolsAndRatings = (character: fullChar) =>
  getPoolsAndRatings(mockStateForCharacter(character), 0)

export const mockGetPenalties = (character: fullChar) =>
  getPenalties(mockStateForCharacter(character), 0)
