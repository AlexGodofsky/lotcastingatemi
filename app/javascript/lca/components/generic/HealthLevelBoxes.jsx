import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from 'material-ui/styles'

import * as calc from '../../utils/calculated'
import { withHealthLevels } from '../../utils/propTypes'

const styles = theme => ({
  boxWrap: {
    display: 'inline-block',
    textAlign: 'center',
    marginRight: '0.25em',
  },
  healthLevelBox: {
    backgroundColor: theme.palette.background.paper,
    width: '1.25em',
    height: '1.25em',
    border: '0.2em solid black'
  },
  healthLevelLabel: { ...theme.typography.caption,
    textAlign: 'center',
  },
  bashingDamage: {
    fontSize: '100%',
    fontWeight: 'bold',
    position: 'relative',
    top: '-0.125em',
    left: '-0.05em',
  },
  lethalDamage: {
    fontSize: '180%',
    fontWeight: '100',
    position: 'relative',
    top: '-0.35em',
    left: '-0.1em',
  },
  aggDamage: {
    fontSize: '150%',
    position: 'relative',
    top: '-0.25em',
    left: '-0.15em',
  },
})

function HealthLevelBoxes(props) {
  const { character, classes } = props
  const totalHealthLevels = calc.totalHealthLevels(character)

  let hlBoxes = []
  let box
  let level

  let aggDamage = character.damage_aggravated
  let lthDamage = character.damage_lethal
  let bshDamage = character.damage_bashing

  let lv0s = character.health_level_0s
  let lv1s = character.health_level_1s
  let lv2s = character.health_level_2s
  let lv4s = character.health_level_4s
  let lvis = character.health_level_incap

  for (var i = 0; i < totalHealthLevels; i++) {
    if (lv0s > 0) {
      level = '0'
      lv0s--
    } else if (lv1s > 0) {
      level = '-1'
      lv1s--
    } else if (lv2s > 0) {
      level = '-2'
      lv2s--
    } else if (lv4s > 0) {
      level = '-4'
      lv4s--
    } else if (lvis > 0) {
      level = 'in'
      lvis--
    } else {
      level = 'X'
    }

    if (aggDamage > 0) {
      box = <div className={ classes.boxWrap } key={ i }>
        <div className={ classes.healthLevelBox }>
          <span className={ classes.aggDamage }>✱</span>
        </div>
        <div className={ classes.healthLevelLabel }>
          { level }
        </div>
      </div>
      aggDamage--
    } else if (lthDamage > 0) {
      box = <div className={ classes.boxWrap } key={ i }>
        <div className={ classes.healthLevelBox }>
          <span className={ classes.lethalDamage }>❌</span>
        </div>
        <div className={ classes.healthLevelLabel }>
          { level }
        </div>
      </div>
      lthDamage--
    } else if (bshDamage > 0) {
      box = <div className={ classes.boxWrap } key={ i }>
        <div className={ classes.healthLevelBox }>
          <span className={ classes.bashingDamage }>╲</span>
        </div>
        <div className={ classes.healthLevelLabel }>
          { level }
        </div>
      </div>
      bshDamage--
    } else {
      box = <div className={ classes.boxWrap } key={ i }>
        <div className={ classes.healthLevelBox }>
          &nbsp;
        </div>
        <div className={ classes.healthLevelLabel }>
          { level }
        </div>
      </div>
    }

    hlBoxes.push(box)
  }

  return <div>
    { hlBoxes }
  </div>
}

HealthLevelBoxes.propTypes = {
  character: PropTypes.shape(withHealthLevels),
  classes: PropTypes.object,
}

export default withStyles(styles)(HealthLevelBoxes)
