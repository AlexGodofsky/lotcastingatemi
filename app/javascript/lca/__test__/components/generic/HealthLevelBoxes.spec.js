import React from 'react'
import renderer from 'react-test-renderer'

import HealthLevelBoxes from '../../../components/generic/HealthLevelBoxes.jsx'
import { mockCharacter } from '../../_mocks'

describe('HealthLevelBoxes', () => {
  it('renders at all', () => {
    const component = renderer.create(
      <HealthLevelBoxes character={ mockCharacter } />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
