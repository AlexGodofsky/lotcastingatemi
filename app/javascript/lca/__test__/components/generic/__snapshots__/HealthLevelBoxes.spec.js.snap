// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthLevelBoxes renders at all 1`] = `
<div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      0
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      -1
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      -1
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      -2
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      -2
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      -4
    </div>
  </div>
  <div
    className="HealthLevelBoxes-boxWrap-1"
  >
    <div
      className="HealthLevelBoxes-healthLevelBox-2"
    >
       
    </div>
    <div
      className="HealthLevelBoxes-healthLevelLabel-3"
    >
      in
    </div>
  </div>
</div>
`;
